src/AdminAan/SFA.DAS.AdminAan.UnitTests/Application/CalendarEvents/Queries/GetCalendarEvent/GetCalendarEventQueryHandlerTests.cs

```diff
@@ -1,26 +1,104 @@
-﻿using AutoFixture.NUnit3;
-using FluentAssertions;
+﻿using FluentAssertions;
 using Moq;
+using RestEase;
 using SFA.DAS.AdminAan.Application.CalendarEvents.Queries.GetCalendarEvent;
+using SFA.DAS.AdminAan.Application.Regions.Queries.GetRegions;
+using SFA.DAS.AdminAan.Application.Schools.Queries;
 using SFA.DAS.AdminAan.Infrastructure;
-using SFA.DAS.Testing.AutoFixture;
 
 namespace SFA.DAS.AdminAan.UnitTests.Application.CalendarEvents.Queries.GetCalendarEvent;
+
 public class GetCalendarEventQueryHandlerTests
 {
-    [Test, MoqAutoData]
-    public async Task Handle_ReturnCalendarEvents(
-        [Frozen] Mock<IAanHubRestApiClient> apiClient,
-        GetCalendarEventQueryHandler handler,
-        GetCalendarEventQueryResult expected,
-        Guid requestedByMemberId,
-        Guid calendarEventId,
-        CancellationToken cancellationToken)
+    private const int RegionId = 1;
+    private const string RegionName = "West";
+    private const long Urn = 123456;
+    private const string SchoolName = "School name";
+
+    [TestCase(null, null)]
+    [TestCase(RegionId, RegionName)]
+    public async Task Handle_ReturnCalendarEvents_WithRegionName(int? regionId, string? regionName)
+    {
+        var cancellationToken = new CancellationToken();
+        var aanHubRestApiClientMock = new Mock<IAanHubRestApiClient>();
+        var referenceDataApiClient = new Mock<IReferenceDataApiClient>();
+        var expected = new GetCalendarEventQueryResult();
+        var requestedByMemberId = Guid.NewGuid();
+        var calendarEventId = Guid.NewGuid();
+        var regions = new List<Region>
+        {
+            new(RegionId, RegionName, 1),
+            new(2, "East", 2)
+        };
+        var regionsResult = new GetRegionsQueryResult(regions);
+
+        expected.CalendarEventId = calendarEventId;
+        expected.RegionId = regionId;
+
+
+        aanHubRestApiClientMock.Setup(x => x.GetRegions(cancellationToken)).ReturnsAsync(regionsResult);
+        aanHubRestApiClientMock.Setup(x => x.GetCalendarEvent(requestedByMemberId, calendarEventId, cancellationToken))
+            .ReturnsAsync(expected);
+
+        var handler = new GetCalendarEventQueryHandler(aanHubRestApiClientMock.Object, referenceDataApiClient.Object);
+        var query = new GetCalendarEventQuery(requestedByMemberId, calendarEventId);
+
+        var actual = await handler.Handle(query, cancellationToken);
+        actual.Should().Be(expected);
+        actual?.RegionName.Should().Be(regionName);
+        var callCount = regionId == null ? 0 : 1;
+        aanHubRestApiClientMock.Verify(x => x.GetRegions(cancellationToken), Times.Exactly(callCount));
+        aanHubRestApiClientMock.Verify(x => x.GetCalendarEvent(requestedByMemberId, calendarEventId, cancellationToken), Times.Once);
+        referenceDataApiClient.Verify(x => x.GetSchoolFromUrn(It.IsAny<string>(), It.IsAny<int>(), It.IsAny<CancellationToken>()), Times.Never);
```

Comment:
Test should avoid verifications which are out of scope. Here we are testing for region name so only that should be verified.